ns1/rest/zones.py

```diff
@@ -95,3 +101,19 @@ def search(self, zone, q=None, has_geo=False, callback=None, errback=None):
             callback=callback,
             errback=errback,
         )
+
+
+# successive pages just extend the list of zones
+def zone_list_pagination(curr_json, next_json):
+    # avoid mutating our inputs. not an issue now, but so it doesn't become one
```

Comment:
why do you think this is going to become an issue?

Reply:
it's for twisted, where you're mainly stacking callbacks, and each callback gets the results of the last one, and our first response gets mutated. It's "not an issue" because the behavior is (happens to be?) correct regardless, but it's a pain for tests and debugging. Confusing to keep track of when you're digging into a mess of callbacks

That said, I should take another whack at this and see if I can make it the twisted transport's problem and keep these functions from having to care.